net/calendar-contacts-server/Portfile

```diff
@@ -0,0 +1,535 @@
+        ${prefix}/etc/${startupitem.location}/org.macports.calendarserver_proxy/org.macports.calendarserver_proxy.plist
+
+    # TLS PFS
+    if ![file exists \
+        ${calendarserverdir}/Library/CalendarServer/etc/dhparam.pem] {
+        # create a shorter, faster DH parameter file for the default installation
+        system -W ${calendarserverdir}/Library/CalendarServer/etc \
+            "openssl dhparam -out dhparam.pem 2048"
+    }
+
+    fs-traverse f_or_d \
+        ${calendarserverdir} {
+        if { [file type ${f_or_d}] != "link" } {
+            file attributes ${f_or_d} \
+                 -owner ${calendarserverUser} -group _calendar
+        }
+    }
+}
+
+notes "Apple Calendar and Contacts Server is a standards-compliant
+server implementing the CalDAV and CardDAV protocols. Full
+deployment requires a working mail server, DNS configuration on both the
+LAN and the internet, including SPF and DKIM records, trusted TLS
+certificates, port forwarding, possibly a mail relay, and more.
+
+Users must reconfigure this installation for their own system, network,
+and security model specifics by editing all necessary files and checking
+file permissions. A subset of these settings are visible in the files:
+
+        port contents calendar-contacts-server
+        port file calendar-contacts-server
+
+These are the locations and network settings for the default configuration:
+
+    Calender and Contacts Server:
+        ${prefix}/var/calendarserver/Library/CalendarServer/Config/calendarserver.plist
+
+    nginx Reverse Proxy:
+        ${prefix}/var/calendarserver/Library/CalendarServer/etc/nginx.conf
+        ports: 8008, 8443, 8800, 8843
+
+A working Calendar and Contacts Server will allow local account authentication
+at these web pages (ports 8008 and 8800 are unencrypted):
+
+        http://${fullhost}:8008
+        https://${fullhost}:8443
+        http://${fullhost}:8800
+        https://${fullhost}:8843
+
+It is necessary to set the active PostgreSQL binaries to postgresql96 using
+the command:
+
+        sudo port select --set postgresql postgresql96
+
+Known issues:
+    * All local accounts have access to calendarserver's password using
+          security find-generic-password -a calendarserver@${fullhost} -g
+      This is a hack to provide twistedcaldav authentication access. See:
+      https://github.com/apple/ccs-calendarserver/blob/master/twistedcaldav/util.py
+      An alternate approach must be used if non-trusted local accounts
+      exist on the Calendar and Contacts Server.
+
+    * There is no destroot capability in ccs-calendarserver; all in.
+
+    * The exisiting Portfile implementation uses native macOS Python,
+      not MacPorts Python.
+
+    * The package ccs-calendarserver is written in Python 2.7, which will
+      reach the end of its life on January 1st, 2020. A future version of
+      pip will drop support for Python 2.7.
```

Comment:
One more reason to stick with MacPorts python.

Reply:
You realize all Python 2.7 will EOL in four months, including MacPorts, right? This isn’t a MacPorts issue, it’s an upstream issue for everyone. Python will **cease security upgrades** to 2.7

Reply:
The goal is to run a python binary under active development with security updates, whether it's native macOS or MacPorts. By definition that's not going to be `python-2.7`. One hopes simply using `six` will provide this transparently with legacy 2.7 scripts, though I haven't tested `six` on this upstream.